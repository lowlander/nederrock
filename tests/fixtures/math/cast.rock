Let X be "123.45"
say X

Cast X
    (X now contains the numeric value 123.45)
say X

Let X be "ff"
say X

Cast X with 16
    (X now contains the numeric value 255 - OxFF)
say X

Cast "12345" into result
    (result now contains the number 12345)
say result

Cast "aa" into result with 16
    (result now contains the number 170 - 0xAA)
say result

Cast 65 into result
    (result now contains the string "A" - ASCII code 65)
say result

Cast 1046 into result
    (result now contains the Cyrillic letter "Ð–" - Unicode code point 1046)
say result
