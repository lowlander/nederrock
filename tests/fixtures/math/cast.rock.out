123.45
123.45
ff
255
12345
170
A
Ð–
